server/src/main/java/org/opensearch/cluster/routing/RoutingNodes.java

```diff
@@ -418,6 +422,29 @@ public ShardRouting activeReplicaWithOldestVersion(ShardId shardId) {
             .orElse(null);
     }
 
+    /**
+     * Returns one active replica shard on a remote node for the given shard id or <code>null</code> if
+     * no such replica is found.
+     * <p>
+     * Since we aim to continue moving forward during remote store migration, replicas already migrated to remote nodes
+     * are preferred for primary promotion
+     */
+    public ShardRouting activeReplicaOnRemoteNode(ShardId shardId) {
+        List<ShardRouting> replicaShardsOnRemote = assignedShards(shardId).stream()
+            .filter(shr -> !shr.primary() && shr.active())
+            .filter((shr) -> {
+                RoutingNode nd = node(shr.currentNodeId());
+                return (nd != null && nd.node().isRemoteStoreNode());
+            })
+            .collect(Collectors.toList());
+        ShardRouting replicaShard = null;
+        if (replicaShardsOnRemote.isEmpty() == false) {
+            Random rand = Randomness.get();
+            replicaShard = replicaShardsOnRemote.get(rand.nextInt(replicaShardsOnRemote.size()));
+        }
+        return replicaShard;
```

Comment:
Do we need to pick active replica with highest checkpoint rather than choosing a random one

Reply:
Checkpoint information is not available at `unassignPrimaryAndPromoteActiveReplicaIfExists`, so we are picking one active replica randomly.